Leanplum-SDK/Classes/Managers/Networking/LPFileTransferManager.m

```diff
@@ -71,7 +99,7 @@ - (id)init
 
         if (_engine == nil) {
             if (!_requestHeaders) {
-                _requestHeaders = [[LPRequestSender sharedInstance] createHeaders];
+                _requestHeaders = [self.requestSender createHeaders];
```

Comment:
This has risk on race condition and thread safety, when multiple calls are made.

Reply:
What's the risk? Since the method is idempotent, I don't understand what the issue is. Please could you elaborate.